### PR TITLE
Define github-hostname for trusted teams

### PR DESCRIPTION
**Pull request by zkdev**
Useful for scenarios where trusted-teams from multiple github instances have to be considered.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Comment by zkdev** [comment]
/hold

do not merge (yet), requires cfg changes

File: model/concourse.py [hunk 1]
```diff
@@ -385,6 +386,12 @@ def secret_name_from_team(team_name: str, key_generation: int) -> str:
         return f'{cfg_name_from_team(team_name)}-{key_generation}'
 
 
+@dataclasses.dataclass(frozen=True)
+class TrustedTeams:
+    github_hostname: str
+    teams: list[str]
```
Comment (ccwienk) [inline model/concourse.py:392]: list is mutable. declaring the dataclass as frozen is inconsistent w/ that. suggestion: declare the class as mutable (non-frozen)

File: model/concourse.py [hunk 2]
```diff
@@ -455,11 +462,31 @@ def expose_deployed_pipelines(self) -> bool:
...
         return self.raw.get('expose_pipelines', True)
 
-    def trusted_teams(self) -> typing.Iterable[str]:
+    def trusted_teams(self) -> collections.abc.Iterable[TrustedTeams]:
         '''Pull requests created/synchronized by members of this team will automatically have
         the required label set by the webhook-dispatcher
         '''
-        return self.raw.get('trusted_teams', [])
+        return [
+            dacite.from_dict(
```
Comment (ccwienk) [inline model/concourse.py:470]: can we keep this simpler and just keep a list of strings? it should be fairly simple to discriminate between qualified and non-qualified teams, as github does not allow `/` (or `.` for that matter) characters in team-names.

Hence, entries of the form `foo/bar` (two parts if splitting by `/`) should be safe to translate to `<org>/<teamname>`, whereas entries of the form `host.tld/foo/bar` (three parts if splitting by `/`) should be safe to interpret as qualified

**Review by ccwienk — CHANGES_REQUESTED** [review]

**Comment by zkdev** [comment]
merge with https://github.com/gardener/cc-utils/pull/1070 , let's follow-up there

**Pull request closed without merging**